### PR TITLE
Add "fallback" option

### PR DESCRIPTION
**Pull request by FrankYFTang**
Address https://github.com/tc39/proposal-intl-displaynames/issues/23

### CODE REVIEW

**Review by ljharb — APPROVED** [review]
Seems reasonable; altho it might be useful to think about what the option would look like if in the future support was added for providing a custom fallback callback.

**Review by gibson042 — APPROVED** [review]

File: spec.emu [hunk 1]
```diff
@@ -316,6 +327,7 @@ contributors: Google, Ecma International
       <li>[[Locale]] is a String value with the language tag of the locale whose localization is used for formatting.</li>
       <li>[[Style]] is one of the String values `"narrow"`, `"short"`, or `"long"`, identifying the display names style used.</li>
       <li>[[Type]] is one of the String values `"language"`, `"region"`, `"script"`, `"currency"`, `"dateField"`, or `"dateSymbol"`, identifying the type of the display names requested.</li>
+      <li>[[Substitute]] is one of the String values `"fallback"`, or `"none"`, identifying what to return while the system does not have the requested display name.</li>
```
Comment (sffc) [inline spec.emu]: Have you done research on precedent in the web platform on possible naming for this sort of feature?  The first thing that comes to mind is multiple load paths for modules.

The words "substitute" and "fallback" seem too redundant to me, but that might be my bias as an ICU developer.  Maybe something more specific like `substitute: "code"` or `substitute: "use-input"`, or use one of those keys with `fallback` as the key?

**Review by sffc — CHANGES_REQUESTED** [review]

File: spec.emu (hunk 1 above)
Comment (FrankYFTang) [reply spec.emu]: how about substitute: 'fallbackToCode' , 'none'

Comment (sffc) [reply spec.emu]: "fallback" is [a noun](https://www.merriam-webster.com/dictionary/fallback). When used as a verb, there are two words. Also, as mentioned in #29, we use kebab case for strings. So, the option you proposed would be, `substitute: "fall-back-to-code"`. That could work, but it is a bit wordy.

Comment (gibson042) [reply spec.emu]: I somehow completely missed 49f62f0afd79b367981a0f6624257f2293416d82 and have been operating from the perspective of `Intl.DisplayNames.prototype.of( codes )`. With scalar input, is there even sufficient value in making this configurable as opposed to always returning undefined? I would expect fallback to be implemented by user code like `codes.map(code => displayNameFormatter.of(code) || code)`.

But if we do want to keep it and are bikeshedding, I'd propose `fallback: "code"` and `fallback: "none"`.

Comment (sffc) [reply spec.emu]: > With scalar input, is there even sufficient value in making this configurable as opposed to always returning undefined?

My positions recorded in the notes still stand in favor of the option.

> I'd propose fallback: "code" and fallback: "none".

LGTM

**Comment by FrankYFTang** [comment]
PTAL I change to 
```
fallback: "code" and fallback: "none".
```

**Comment by FrankYFTang** [comment]
Everyone PTAL by end of Friday May 17 2019. Thanks

File: spec.emu [hunk 2]
```diff
@@ -248,7 +250,12 @@ contributors: Google, Ecma International
         1. If the result of ! IsValidCodeForDisplayNames(_displayNames_.[[Type]], _code_) is *false*, throw a *RangeError* exception.
         1. Let _fields_ be _displayNames_.[[Fields]].
         1. Let _name_ be _fields_[[_code_]].
-        1. Return _name_.
+        1. If _name_ is not *undefined*, then
+          1. Return _name_.
+        1. Let _fallback_ be _displayNames_.[[Fallback]].
```
Comment (ljharb) [inline spec.emu]: seems like the `_fallback_` var isn't really needed, could do `If _displayNames_.[[Fallback]] is`?

File: spec.emu [hunk 3]
```diff
@@ -316,6 +327,7 @@ contributors: Google, Ecma International
       <li>[[Locale]] is a String value with the language tag of the locale whose localization is used for formatting.</li>
       <li>[[Style]] is one of the String values `"narrow"`, `"short"`, or `"long"`, identifying the display names style used.</li>
       <li>[[Type]] is one of the String values `"language"`, `"region"`, `"script"`, `"currency"`, `"dateField"`, or `"dateSymbol"`, identifying the type of the display names requested.</li>
+      <li>[[Fallback]] is one of the String values `"code"`, or `"none"`, identifying the fallback return while the system does not have the requested display name.</li>
```
Comment (ljharb) [inline spec.emu]:
```suggestion
      <li>[[Fallback]] is one of the String values `"code"`, or `"none"`, identifying the fallback return when the system does not have the requested display name.</li>
```

**Review by ljharb — APPROVED** [review]

**Review by littledan — APPROVED** [review]

**Comment by littledan** [comment]
This seems to match our conclusion in the ECMA-402 meeting.
